components/layout_thread/lib.rs

```diff
@@ -1643,6 +1659,11 @@ impl LayoutThread {
     }
 }
 
+impl<'a> ProfilerMetadataFactory for &'a LayoutThread {
```

Comment:
This looks like it can be `impl ProfilerMetadataFactory for LayoutThread` instead.